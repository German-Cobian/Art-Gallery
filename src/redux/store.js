import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import artworksReducer from './artworksApi';

const reducers = combineReducers({
  artworksReducer,
});

const store = createStore(
  reducers,
  applyMiddleware(logger, thunk),
);

export default store;
