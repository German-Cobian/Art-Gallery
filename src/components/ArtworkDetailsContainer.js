import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router';
import { fetchArtworks } from '../redux/artworksApi';
import ArtworkDetails from './ArtworkDetails';
/* eslint-disable eqeqeq */

const ArtworkDetailsContainer = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchArtworks());
  }, []);

  const artworksState = useSelector((state) => state.artworksReducer);
  const { id } = useParams();

  const filteredArtworks = artworksState.filter((artwork) => artwork.id == id);

  return (
    <div className="artworkdetails">
      { filteredArtworks.map((item) => (
        <ArtworkDetails
          key={item.id}
          id={item.id}
          image={item.image_id}
          title={item.title}
          type={item.artwork_type_title}
          medium={item.medium_display}
          dimensions={item.dimensions}
          date={item.date_display}
          artist={item.artist_title}
          from={item.place_of_origin}
          category={item.department_title}
        />
      ))}
    </div>
  );
};

export default ArtworkDetailsContainer;
