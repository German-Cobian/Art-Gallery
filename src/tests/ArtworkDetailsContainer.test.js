import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom/extend-expect';
import ArtworkDetailsContainer from '../components/ArtworkDetailsContainer';
import store from '../redux/store';

describe('ArtworkDetailsContainer component', () => {
  test('matches ArtworkDetailsContainer component snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <ArtworkDetailsContainer />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
