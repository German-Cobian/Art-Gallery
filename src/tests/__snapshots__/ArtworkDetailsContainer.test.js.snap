// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtworkDetailsContainer component matches ArtworkDetailsContainer component snapshot 1`] = `
<div
  className="artworkdetails"
/>
`;
